### PR TITLE
Handle supported algorithms and curves

### PR DESCRIPTION
**Pull request by cedelavergne-ledger**
Update the App to fix some issues arround supported Algorithms and Curves

- Fix `nistp256` template
- Disable `RSA4096` (today a watchdog issue prevent to use it in Seed Mode)
- Update the supported `ECC`, updating also few wordings
- Update the documentation
- Update version to **2.1.0**
- Update manual tool to provide `expert` mode, allowing to setup the keys templates 

# Checklist

- [X] App update process has been followed 
- [X] Target branch is `develop` 
- [X] Application version has been bumped

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -122,6 +122,7 @@ DISABLE_STANDARD_WEBUSB = 1
 DEFINES   += CUSTOM_IO_APDU_BUFFER_SIZE=\(255+5+64\)
 DEFINES   += HAVE_USB_CLASS_CCID
 DEFINES   += HAVE_RSA
+# DEFINES   += WITH_SUPPORT_RSA4096
```
Comment (xchapron-ledger) [inline Makefile:125]: Can you add a comment explaining why it's not enabled for now?

**Review by xchapron-ledger — APPROVED** [review]

File: src/gpg_data.c [hunk 2]
```diff
@@ -832,18 +858,39 @@ int gpg_apdu_get_key_data(unsigned int ref) {
 
     // encrypted part
     switch (keygpg->attributes.value[0]) {
-        case KEY_ID_RSA:  // RSA
+        case KEY_ID_RSA:
+            ksz = U2BE(G_gpg_vstate.mse_dec->attributes.value, 1) >> 3;
```
Comment (tdejoigny-ledger) [inline src/gpg_data.c:862]: Why we need to /8 here ?

File: src/gpg_data.c [hunk 3]
```diff
@@ -937,26 +985,51 @@ int gpg_apdu_put_key_data(unsigned int ref) {
                 break;
             }
             offset = G_gpg_vstate.io_offset;
+            ksz = U2BE(G_gpg_vstate.mse_dec->attributes.value, 1) >> 3;
```
Comment (tdejoigny-ledger) [inline src/gpg_data.c:988]: same comment as above

**Review by tdejoigny-ledger — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (cedelavergne-ledger) [reply Makefile:126]: Done

File: src/gpg_data.c (hunk 2 above)
Comment (cedelavergne-ledger) [reply src/gpg_data.c:862]: Legacy code done like this.
Most probably the size needed is in bytes, whereas in the descriptors it is in bits.

File: src/gpg_data.c (hunk 3 above)
Comment (cedelavergne-ledger) [reply src/gpg_data.c:988]: same

**Pull request merged**